src/Mvc/Mvc.ViewFeatures/src/Rendering/HtmlHelperComponentExtensions.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Components;
+using Microsoft.AspNetCore.Html;
+using Microsoft.AspNetCore.Mvc.ViewFeatures;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.Mvc.Rendering
+{
+    /// <summary>
+    /// Extensions for rendering components.
+    /// </summary>
+    public static class HtmlHelperComponentExtensions
+    {
+        /// <summary>
+        /// Renders the <typeparamref name="TComponent"/> <see cref="IComponent"/>.
+        /// </summary>
+        /// <param name="htmlHelper">The <see cref="IHtmlHelper"/>.</param>
+        /// <param name="renderMode">The <see cref="RenderMode"/> for the component.</param>
+        /// <returns>The HTML produced by the rendered <typeparamref name="TComponent"/>.</returns>
+        public static Task<IHtmlContent> RenderComponentAsync<TComponent>(this IHtmlHelper htmlHelper, RenderMode renderMode) where TComponent : IComponent
+            => RenderComponentAsync<TComponent>(htmlHelper, renderMode, parameters: null);
+
+        /// <summary>
+        /// Renders the <typeparamref name="TComponent"/> <see cref="IComponent"/>.
+        /// </summary>
+        /// <param name="htmlHelper">The <see cref="IHtmlHelper"/>.</param>
+        /// <param name="parameters">An <see cref="object"/> containing the parameters to pass
+        /// to the component.</param>
+        /// <param name="renderMode">The <see cref="RenderMode"/> for the component.</param>
+        /// <returns>The HTML produced by the rendered <typeparamref name="TComponent"/>.</returns>
+        public static Task<IHtmlContent> RenderComponentAsync<TComponent>(
+            this IHtmlHelper htmlHelper,
+            RenderMode renderMode,
+            object parameters) where TComponent : IComponent
+        {
+            if (htmlHelper is null)
+            {
+                throw new ArgumentNullException(nameof(htmlHelper));
+            }
+
+            var viewContext = htmlHelper.ViewContext;
+            var componentRenderer = viewContext.HttpContext.RequestServices.GetRequiredService<IComponentRenderer>();
+            return componentRenderer.RenderComponentAsync(viewContext, typeof(TComponent), renderMode, parameters);
+        }
+    }
```

Comment:
We should also add non-generic overloads. The HTML helper is a little too limiting.

Reply:
I thought about it, but in general we wouldn't encourage users to use the html helper if there's an equally capable tag helper available. Do you feel strongly about adding the additional overload?

Reply:
I don't really care that much about the TagHelper, but it should be uber cheap to add this, so I'm with @rynowak on this. There's not even need to add or change tests for them, simply chain the implementations. I trust you know what to do :)